### DISCUSSION TITLE
[lodash] upperFirst stronger return type

### DISCUSSION THREAD

**Discussion by basti-n** [DefinitelyTyped/DefinitelyTyped#68489] [Issues with a @types package] [answered]
Currently `upperFirst` only returns a string, not leveraging the power of `Capitalize`.

```js
upperFirst(string?: string): string;
````

could become:

```js
upperFirst<S extends string>(string?: S): Capitalize<S>;
```

Any opinions? 

For proposed change, see: :https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68488/files

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
Sounds like a great idea, feel free to send the PR

**Comment by aj-r** [comment]
Should we use Uncapitalize for lowerFirst, too? Similar for toUpper and toLower.

**Reply by hkleungai** [reply]
Hi, since the original PR has not yet been updated for months, I have opened [another PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70570) for that. 

Please help review when you have time.

**Reply by basti-n** [reply]
lgtm, thanks for picking up my work. There will be plenty of projects happy with the change 👏
